src/index.ts

```diff
@@ -114,20 +115,24 @@ export class Telnet extends EventEmitter {
   private response: string[] = undefined
   private socket: Socket | null = null
   private state: TelnetState = null
+  private decoder: StringDecoder = null
 
   constructor() {
     super()
 
     this.on('data', data => this.pushNextData(data))
     this.on('end', () => {
+      const remaining = this.decoder.end()
+      if (remaining)
```

Comment:
Please place a newline before `if`. Improves readability. I'm not a fan of this new condensed style everyone's using these days. :)

Reply:
I'm not really sure what you mean by that. Can you explain?

Reply:
Sorry, it’s a nit-pick, I like having an empty line before ‘if’.